plugin/src/main/java/org/opensearch/ml/rest/mcpserver/RestMLMcpToolsRegisterAction.java

```diff
@@ -73,33 +79,42 @@ public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client
         }
         ActionRequestValidationException exception = new ActionRequestValidationException();
         MLMcpToolsRegisterNodesRequest registerNodesRequest = getRequest(request);
-        if (CollectionUtils.isEmpty(registerNodesRequest.getMcpTools().getTools())) {
+        if (CollectionUtils.isEmpty(registerNodesRequest.getMcpTools())) {
```

Comment:
`private List<McpTool> tools;`, rather than declaring tools as a list, can this be Set? then we don't need to do duplicate check by ourselves?

Reply:
Do you mean check the duplicate names? The `List` is at `McpToolRegisterInput` level and we're checking the name of the object, changing it to set won't help.